features/FEATURE_LWIP/lwip-interface/lwip_stack.c

```diff
@@ -104,7 +104,7 @@ static void mbed_lwip_socket_callback(struct netconn *nc, enum netconn_evt eh, u
 /* TCP/IP and Network Interface Initialisation */
 static struct netif lwip_netif;
 static bool lwip_dhcp = false;
-static char lwip_mac_address[NSAPI_MAC_SIZE] = "\0";
+static char lwip_mac_address[NSAPI_MAC_SIZE + 1] = "\0";
```

Comment:
Also, while touching this,  that existing `="\0"` is bizarre, and adds to the terminator confusion. Just `=""` or no initialiser at all would do.